source/extensions/resource_monitors/downstream_connections/downstream_connections_monitor.cc

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "extensions/resource_monitors/downstream_connections/downstream_connections_monitor.h"
+
+#include <thread>
+
+#include "envoy/config/resource_monitor/downstream_connections/v3/downstream_connections.pb.h"
+
+#include "common/common/assert.h"
+
+// tbd
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace DownstreamConnectionsMonitor {
+
+DownstreamConnectionsMonitor::DownstreamConnectionsMonitor(
+    const envoy::config::resource_monitor::fixed_heap::v3::DownstreamConnectionsConfig& config)
+    : max_downstream_conns_(config.max_downstream_connections()),
+      global_total_downstream_conns_(0) {
+  ASSERT(max_downstream_conns_ > 0);
+}
+
+void DownstreamConnectionsMonitor::updateResourceUsage(
+    Server::ResourceMonitor::Callbacks& callbacks) {
+  Server::ResourceUsage usage;
+  global_total_downstream_conns_.exchange(0);
+  for (auto& downstream_conns_ : thread_local_total_downstream_conns_) {
```

Comment:
nit: You need a mutex to protect against concurrent read and modification to the thread_local_total_downstream_conns_ map.